src/js/libcs/OpDrawing.js

```diff
@@ -106,6 +106,183 @@ evaluator.fillcircle$2 = function(args, modifs) {
     return eval_helper.drawcircle(args, modifs, "F");
 };
 
+evaluator.drawarc$3 = function(args, modifs) {
+    return eval_helper.drawarc(args, modifs, "D");
+};
+
+evaluator.fillarc$3 = function(args, modifs) {
+    return eval_helper.drawarc(args, modifs, "F");
+};
+
+
+eval_helper.drawarc = function(args, modifs, df) {
+    var v1 = evaluateAndVal(args[0]);
+    var v2 = evaluateAndVal(args[1]);
+    var v3 = evaluateAndVal(args[2]);
+
+    var a = List.normalizeZ(csgeo.csnames[args[0].name].homog);
+    var b = List.normalizeZ(csgeo.csnames[args[1].name].homog);
+    var c = List.normalizeZ(csgeo.csnames[args[2].name].homog);
+    var abcdet = List.det3(a, b, c);
```

Comment:
Why access `csgeo` here? After all, the arguments could already be some user-provided coordinate vector, and not a geometric element at all. Better use `evaluateAndHomog` instead of `evaluateAndVal`, and let that take care of turning the input into homogeneous coordinates in whatever way seems appropriate.